### PR TITLE
Simplify test for AMP plugins

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

unify and parameterized tests for AMP plugins, also adding default for native AMP

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/plugins/test_amp_plugin.py [hunk 1]
```diff
@@ -1,97 +0,0 @@
...
-def test_amp_choice_custom_ddp_cpu(device_count_mock, ddp_backend, gpus):
-
-    class MyNativeAMP(NativeMixedPrecisionPlugin):
-        pass
-
-    trainer = Trainer(
-        precision=16,
-        amp_backend='native',
-        accelerator=ddp_backend,
-        plugins=[MyNativeAMP()],
```
Comment (Borda) [inline tests/plugins/test_amp_plugin.py:38]: here is missing test for defaults

File: tests/plugins/test_amp_plugin.py [hunk 2]
```diff
@@ -1,97 +0,0 @@
...
-        accelerator='ddp_spawn',
-        gpus=2,
-        precision=16,
-        track_grad_norm=2,
-        log_every_n_steps=1,
-    )
-    trainer.fit(model)
-
-
-class UnscaleAccumulateGradBatchesBoringModel(BoringModel):
```
Comment (Borda) [inline tests/plugins/test_amp_plugin.py:71]: this is exactly the same as `GradientUnscaleBoringModel`

File: tests/plugins/test_amp_plugins.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+class GradientUnscaleBoringModel(BoringModel):
+
+    def on_after_backward(self):
+        norm = torch.nn.utils.clip_grad_norm_(self.parameters(), 2)
+        if not (torch.isinf(norm) or torch.isnan(norm)):
+            assert norm.item() < 15.
+
+
+@RunIf(min_gpus=2, amp_native=True)
+@pytest.mark.parametrize('accum', [1, 2])
```
Comment (Borda) [inline tests/plugins/test_amp_plugins.py:63]: the only real diff before was nb accumulated batches, the classes were identical

File: tests/plugins/test_amp_plugins.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        "SLURM_NODEID": "0",
+        "LOCAL_RANK": "0",
+        "SLURM_LOCALID": "0",
+    }
+)
+@mock.patch('torch.cuda.device_count', return_value=2)
+@pytest.mark.parametrize('ddp_backend,gpus', [('ddp', 2), ('ddp2', 2), ('ddp_spawn', 2)])
+@pytest.mark.parametrize(
+    'amp,plugin_arg,plugin_cls', [
+        pytest.param('native', None, NativeMixedPrecisionPlugin, marks=RunIf(amp_native=True)),
```
Comment (Borda) [inline tests/plugins/test_amp_plugins.py:35]: this default is newly added

**Review by carmocca — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
